mixin/smarty-v2@1/mixin.php

```diff
@@ -19,14 +19,9 @@
   }
 
   $register = function() use ($dir) {
-    // This implementation is useful for older versions of CiviCRM. It can be replaced/updated going forward (v1.1+).
-    $smarty = CRM_Core_Smarty::singleton();
-    if (!is_array($smarty->template_dir)) {
-      $this->template_dir = [$smarty->template_dir];
```

Comment:
Fun fact @eileenmcnaughton - I think you're patch here probably removed an edge-case bug that we didn't notice before. :)

The reference to `$this` should be `$smarty`. That's a copy-paste-adapt error.

(Of course, it was asymptomatic because this code-path is hard to get into... but it was a bug!)

Reply:
How did this pop up on your radar again now?

Reply:
@eileenmcnaughton - @jofranz spotted it with a linter (https://github.com/totten/civix/issues/294)

Reply:
Ah -